tasks/install-private-dependencies

```diff
@@ -0,0 +1,276 @@
+const cockpitBranch = process.env.cockpitBranch;
+const cockpitCommit = process.env.cockpitCommit;
+
+/* ------------ Shared functions ------------ */
+
+function ifOKPrintIfErrPrintAndExit(spawnPs){
+  if(spawnPs.status > 0){
+    process.stderr.write(spawnPs.stderr.toString() + '\n');
+    process.exit(1);
+  }
+  // TODO: understand why the following blocks the task logs
+  // else {
+  //   process.stdout.write(spawnPs.stdout.toString() + '\n');
+  // }
+}
+
+/* ------------ Shared functions ------------ */
+
+let out =null;
+
+/* ------------ slide2html ------------ */
+
+//clean
+process.stdout.write('Cleaning slide2html dir...');
+fs.removeSync(slide2htmlDir);
+process.stdout.write('done!\n');
+
+// download
+process.stdout.write('Downloading ' + slide2htmlBranch + ' branch for slide2html...');
+out = spawnSync('git', [
+  'clone',
+  '-b',
+  slide2htmlBranch,
+  'https://' + process.env.CI_USER_TOKEN + ':x-oauth-basic@github.com/ASQ-USI/slide2html.git',
+  slide2htmlDir
+],{
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+// check out specific commit
+process.stdout.write('Checking out commit ' + slide2htmlCommit + ' for slide2html...');
+out = spawnSync('git', [
+  'checkout',
+  slide2htmlCommit
+],{
+  cwd: slide2htmlDir,
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+// install npm deps
+process.stdout.write('Installing npm dependencies for slide2html...');
+out = spawnSync('npm', [
+  'install',
+  '--unsafe-perm',
+  '--production'
+], {
+  cwd: slide2htmlDir,
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+/* ------------ slide2html ------------ */
+
+/* ------------ ui ------------ */
+
+//clean
+process.stdout.write('Cleaning ui-asq dir...');
+fs.removeSync(uiDir);
+process.stdout.write('done!\n');
+
+// download
+process.stdout.write('Downloading ' + uiBranch + ' branch for ui-asq...');
+out = spawnSync('git', [
+  'clone',
+  '-b',
+  uiBranch,
+  'https://' + process.env.CI_USER_TOKEN + ':x-oauth-basic@github.com/ASQ-USI/ui-asq.git',
+  uiDir
+],{
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+// check out specific commit
+process.stdout.write('Checking out commit ' + uiCommit + ' for ui-asq...');
+out = spawnSync('git', [
+  'checkout',
+  uiCommit
+],{
+  cwd: uiDir,
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+// install npm deps
+process.stdout.write('Installing npm dependencies for ui-asq...');
+out = spawnSync('npm', [
+  'install',
+  '--unsafe-perm'
+], {
+  cwd: uiDir,
+  timeout: timeout
+});
+ifOKPrintIfErrPrintAndExit(out);
+process.stdout.write('done!\n');
+
+// avoid using root .bowerrc by creating a local one
+// with the default settings
+process.stdout.write('Creating local .bowerrc...');
+let bowerrcContents = `{
+  "directory": "bower_components"
```

Comment:
I copied it from your scripts, can you please fix this. I would like to avoid doing a mess.